src/common/xio_connection.c

```diff
@@ -1751,6 +1751,9 @@ static void xio_fin_req_timeout(void *data)
 	/* flush all messages back to user */
 	xio_connection_notify_msgs_flush(connection);
 
+	if (connection->nexus)
+		xio_nexus_close(connection->nexus, &connection->session->observer);
```

Comment:
please note that xio_nexus_close ends up with putting NULL to connection->nexus
so the call to close nexus in xio_connection_post_destroy occur only if connection->nexus != NULL